scheduler-worker/src/main/java/gov/nasa/jpl/aerie/scheduler/worker/services/SynchronousSchedulerAgent.java

```diff
@@ -281,6 +286,24 @@ public void schedule(
     }
   }
 
+  public List<SchedulingActivityDirective> updateEverythingWithNewAnchorIds(Plan solutionPlan, Map<SchedulingActivityDirective, ActivityDirectiveId> instancesToIds){
+    ArrayList<SchedulingActivityDirective> updatedActs = new ArrayList<SchedulingActivityDirective>();
+    var planActs = new HashSet<>(solutionPlan.getActivities());
+    for (SchedulingActivityDirective act : planActs) {
+      if (act.anchorId() != null) {
+        SchedulingActivityDirective actAnchored = solutionPlan.getActivitiesById().get(act.anchorId());
+        SchedulingActivityDirective updatedAct = SchedulingActivityDirective.copyOf(act, new SchedulingActivityDirectiveId(instancesToIds.get(actAnchored).id()), act.anchoredToStart(), act.startOffset());
+        updatedActs.add(updatedAct);
+        solutionPlan.replaceActivity(act, updatedAct);
+        ActivityDirectiveId value = instancesToIds.get(act);
+        instancesToIds.remove(act);
+        instancesToIds.put(updatedAct, value);
+      }
+    }
+    return updatedActs;
+  }
+
+
```

Comment:
do a pass for `final`

Reply:
Done. All local vars were converted to final var